src/llvm/di.rs

```diff
@@ -404,6 +434,59 @@ impl DISanitizer {
 
         replace
     }
+
+    fn di_compile_units(&self) -> Vec<DICompileUnit> {
+        let compile_unit_name = c"llvm.dbg.cu";
+
+        // Get the number of DICompileUnit.
+        let num_di_cu =
+            unsafe { LLVMGetNamedMetadataNumOperands(self.module, compile_unit_name.as_ptr()) };
+
+        // Create a vector to hold them all.
+        let mut di_cus: Vec<LLVMValueRef> = vec![core::ptr::null_mut(); num_di_cu as usize];
+
+        unsafe {
+            LLVMGetNamedMetadataOperands(
+                self.module,
+                compile_unit_name.as_ptr(),
+                di_cus.as_mut_ptr(),
+            )
+        };
+
+        di_cus
+            .into_iter()
+            .map(|v| unsafe { DICompileUnit::from_value_ref(v) })
+            .collect()
+    }
+
+    /// Removes replaced `c_void` Rust enum from all [`DICompileUnit`]. After
+    /// replacing `c_void` enum with a [`DIBasicType`] the [`DICompileUnit`] still
+    /// hold a reference to the enum and still believes it is a [`DICompositeType`]
+    /// which triggers a casting assertion in LLVM.
+    fn fix_di_compile_units(&mut self) {
+        for mut di_cu in self.di_compile_units() {
+            let tmp_cu = di_cu.clone();
+            let enum_types = tmp_cu.enum_types();
+            let mut need_replace = false;
+            let new_enums: Vec<_> = enum_types
+                .into_iter()
+                .filter(|e| {
+                    let kind = unsafe { LLVMGetMetadataKind(e.metadata_ref) };
```

Comment:
Perhaps we should add a safe method `kind()` which returns `LLVMMetadataKind` and hides the `unsafe` call. Let's try to keep `unsafe` only in wrappers.